back/engines/commercial/analytics/app/services/analytics/populate_dimensions_service.rb

```diff
@@ -75,13 +58,31 @@ def populate_dates
         create_dates(from, to)
       end
 
-      def populate_locales
-        locales = AppConfiguration.instance.settings('core', 'locales')
-        locales.each do |locale_name|
-          next if Analytics::DimensionLocale.exists?(name: locale_name)
+      def create_dates(from, to)
+        insert_dates = (from..to).map do |date|
+          {
+            date: date,
+            week: date.beginning_of_week.to_date,
+            month: "#{date.year}-#{date.strftime('%m')}",
+            year: date.year
+          }
+        end
+        if insert_dates.present?
+          insert_dates.each_slice(200) do |dates|
+            Analytics::DimensionDate.insert_all(dates)
+          end
+        end
+      end
 
-          Analytics::DimensionLocale.create!(name: locale_name)
+      def populate_locales
+        locales = AppConfiguration.instance.settings('core', 'locales').map do |locale_name|
+          { name: locale_name }
         end
+
+        current_locales = Analytics::DimensionLocale.all.as_json(only: %i[name])
+        return unless current_locales.to_set != locales.as_json.to_set
```

Comment:
Nit: If feel like `return if current_locales.to_set == locales.as_json.to_set` would be more readable.